graphs/lca.cpp

```diff
@@ -1,35 +1,39 @@
+#include <bits/stdc++.h>
+using namespace std;
 struct lca {
+	using pii = pair<int, int>;
 	int Log;
 	vector<vector<int>> memo;
-	vector<int> depth;
+	vector<int> depth, dist;
 
-	lca(const vector<vector<int>> &graph, int root) {//0 - based nodes
+	// use weights of 1 for unweighted tree
+	lca(const vector<vector<pii>> &graph, int root = 0) {//0 - based nodes
 		int n = graph.size();
 		Log = 1;
 		while((1<<Log) < n) ++Log;
 		depth.resize(n,0);
+		dist.resize(n,0);
 		memo.resize(n,vector<int>(Log));
 		dfs(root,root,graph);
 	}
 
-	void dfs(int node, int par, const vector<vector<int>> &graph) {
+	void dfs(int node, int par, const vector<vector<pii>> &graph) {
 		memo[node][0] = par;
 		for(int i = 1; i < Log; ++i) {
 			memo[node][i] = memo[memo[node][i-1]][i-1];
 		}
-		for(int to : graph[node]) {
+		for(auto [to, w] : graph[node]) {
 			if(to == par) continue;
 			depth[to] = 1 + depth[node];
+			dist[to] = w + dist[node];
 			dfs(to, node, graph);
 		}
 	}
 
 	int kthPar(int node, int k) {
-		for(int bit = 0; bit < Log; ++bit) {
-			if(k&(1<<bit)) {
+		for(int bit = 0; bit < Log; ++bit)
+			if(k&(1<<bit))
 				node = memo[node][bit];
-			}
-		}
```

Comment:
my personal opinion is you shouldn’t re-format code just to match your style when it doesn’t change behavior, or add any value (for example refactor to meet SOLID principals)

the code changes don’t provide much value, and can only possibly introduce bugs

Reply:
This I have never considered. I made the change to help keep the struct from getting too long, but this is a good point. I think this is generally true; however, with CP I think the goal is the shortest template which is still readable. I am not adamant we keep this change, but I will give it it one more shot.